libraries/AP_Param/AP_Param.h

```diff
@@ -542,11 +542,16 @@ class AP_ParamT : public AP_Param
             set(v);
         }
     }
-    
+
     /// Value setter - set value, tell GCS
     ///
     void set_and_notify(const T &v) {
+// We do want to compare each value, even floats, since it being the same here
+// is the result of previously setting it.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
         if (v != _value) {
+#pragma GCC diagnostic pop
```

Comment:
Since this is for scalar types couldn't we use the *is_equal* here? Also, am I seeing it wrong or does the *set_and_save* method below also does a comparison but using the float epsilon? And I'm surprised there's no issue with the *set_and_save_ifchanged* below.